pdr_backend/sim/sim_plotter.py

```diff
@@ -25,277 +20,235 @@ class SimPlotter:
+        # avoid race conditions with the pickling itself
+        if file_age_in_seconds("sim_state/st_final.pkl") < 3:
+            time.sleep(3)
 
-        # col 3 is empty, for overflow of aimodel_varimps's y-axis labels
-        self.ax_aimodel_varimps = fig.add_subplot(gs[:, 4])
+        with open("sim_state/st_final.pkl", "rb") as f:
+            self.st = pickle.load(f)
 
-        self.ax_aimodel_response = fig.add_subplot(gs[:, 5])
+        with open("sim_state/aimodel_plotdata_final.pkl", "rb") as f:
+            self.aimodel_plotdata = pickle.load(f)
 
-        # attributes to help update plots' state quickly
-        self.N: int = 0
-        self.N_done: int = 0
-        self.x: List[float] = []
+        return self.st, "final"
 
-        self.shown_plot_before: bool = False
-        self.computed_plot_before: bool = False
+    def init_state(self):
+        files = glob.glob("sim_state/*")
+        for f in files:
+            os.remove(f)
 
-    # pylint: disable=too-many-statements
-    @enforce_types
-    def compute_plot(
-        self,
-        aimodel_plotdata: AimodelPlotdata,
-        do_show_plot: bool,
-        do_save_plot: bool,
-    ) -> Optional[str]:
-        """
-        @description
-          Create / update whole plot, with many subplots
-
-        @arguments
-          aimodel_plotdata -- has model, X_train, etc
-          do_show_plot -- render on-screen in a window?
-          do_save_plot -- export as png?
-
-        @return
-          img_filename - filename of saved plot (None if not done)
-        """
-        if not self.shown_plot_before:
-            plt.ion()
-            if do_show_plot:
-                plt.show()
-            self.shown_plot_before = True
-
-        # update N, N_done, x. **Update x only after updating N, N_done!**
-        self.N = len(self.st.pdr_profits_OCEAN)
-        self.N_done = len(self.x)  # what # points have been plotted previously
-        self.x = list(range(0, self.N))
-
-        # main work: create/update subplots
-        self._plot_pdr_profit_vs_time()
-        self._plot_trader_profit_vs_time()
-
-        self._plot_accuracy_vs_time()
-        self._plot_f1_precision_recall_vs_time()
-        self._plot_pdr_profit_vs_ptrue()
-        self._plot_trader_profit_vs_ptrue()
-
-        self._plot_aimodel_varimps(aimodel_plotdata)
-        self._plot_aimodel_response(aimodel_plotdata)
-
-        # final pieces of making plot
-        self.fig.set_size_inches(WIDTH, HEIGHT)
-        self.fig.tight_layout(pad=0.5, h_pad=1.0, w_pad=1.0)
-        plt.subplots_adjust(wspace=0.3)
-
-        # save to png?
-        img_filename = None
-        if do_save_plot:
-            img_filename = self.ppss.sim_ss.unique_final_img_filename()
-            savefig(img_filename)
-
-        # wrapup for reloop
-        if do_show_plot:
-            plt.pause(0.001)
-        self.computed_plot_before = True
-
-        return img_filename
```

Comment:
I verified this moved to sim_plots and based on the plots rendered, I think it checks out